go/cmd/dolt/commands/status.go

```diff
@@ -63,215 +96,574 @@ func (cmd StatusCmd) ArgParser() *argparser.ArgParser {
+			if err != nil {
+				return nil, err
+			}
+
+			// determine if the table should be ignored
+			ignored, err := ignorePatterns.IsTableNameIgnored(tableName)
+			if conflict := doltdb.AsDoltIgnoreInConflict(err); conflict != nil {
+				ignoredTables.Conflicts = append(ignoredTables.Conflicts, *conflict)
+			} else if err != nil {
+				return nil, err
+			} else if ignored == doltdb.DontIgnore {
+				ignoredTables.DontIgnore = append(ignoredTables.DontIgnore, tableName)
+			} else if ignored == doltdb.Ignore {
+				ignoredTables.Ignore = append(ignoredTables.Ignore, tableName)
+			} else {
+				return nil, fmt.Errorf("unrecognized ignore result value: %v", ignored)
+			}
+			if err != nil {
+				return nil, err
+			}
+			shouldIgnoreTable := ignored == doltdb.Ignore
+
+			switch status {
+			case "renamed":
+				// for renamed tables, add both source and dest changes
+				parts := strings.Split(tableName, " -> ")
+				srcTableName := parts[0]
+				dstTableName := parts[1]
+
+				if workingTableNames[dstTableName] {
+					unstagedTables[srcTableName] = "deleted"
+					untrackedTables[dstTableName] = "new table"
+				} else if stagedTableNames[dstTableName] {
+					stagedTables[tableName] = status
+				}
+			case "conflict":
+				conflictsPresent = true
+				if isStaged {
+					stagedTables[tableName] = status
+				} else {
+					unmergedTables[tableName] = "both modified"
+				}
+			case "deleted", "modified", "added", "new table":
+				if shouldIgnoreTable {
+					continue
+				}
+
+				if isStaged {
+					stagedTables[tableName] = status
+				} else {
+					isTableConflicted := conflictedTables[tableName]
+					if !isTableConflicted {
+						if status == "new table" {
+							untrackedTables[tableName] = status
+						} else {
+							unstagedTables[tableName] = status
+						}
+					}
+				}
+			case "schema conflict":
+				conflictsPresent = true
+				schemaConflictTables[tableName] = true
+			default:
+				panic(fmt.Sprintf("table %s, unexpected merge status: %s", tableName, status))
+			}
+		}
 	}
-	headHash, err := headCommit.HashOf()
-	if err != nil {
-		return err
+
+	pd := printData{
+		branchName:                branchName,
+		remoteName:                remoteName,
+		remoteBranchName:          remoteBranchName,
+		ahead:                     ahead,
+		behind:                    behind,
+		conflictsPresent:          conflictsPresent,
+		showIgnoredTables:         showIgnoredTables,
+		statusPresent:             statusPresent,
+		mergeActive:               mergeActive,
+		stagedTables:              stagedTables,
+		unstagedTables:            unstagedTables,
+		untrackedTables:           untrackedTables,
+		unmergedTables:            unmergedTables,
+		ignoredTables:             ignoredTables,
+		constraintViolationTables: constraintViolationTables,
+		schemaConflictTables:      schemaConflictTables,
+		ignorePatterns:            ignorePatterns,
+		dataConflictTables:        dataConflictTables,
```

Comment:
This function is 200 lines long. Possible to break into bite size chunks?

Reply:
Splitting out most of the "gathering" logic into smaller functions.